go/libraries/doltcore/sqle/enginetest/dolt_queries.go

```diff
@@ -90,76 +90,111 @@ var ViewsWithAsOfScriptTest = queries.ScriptTest{
 		},
-		{
-			Query: "show create table a as of @Commit1;",
-			Expected: []sql.Row{
-				{"a", "CREATE TABLE `a` (\n" +
-					"  `pk` int NOT NULL,\n" +
-					"  `c1` int,\n" +
-					"  PRIMARY KEY (`pk`)\n" +
-					") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin",
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:       "show create table a as of @Commit0;",
+				ExpectedErr: sql.ErrTableNotFound,
+			},
+			{
+				Query: "show create table a as of @Commit1;",
+				Expected: []sql.Row{
+					{"a", "CREATE TABLE `a` (\n" +
+						"  `pk` int NOT NULL,\n" +
+						"  `c1` int,\n" +
+						"  PRIMARY KEY (`pk`)\n" +
+						") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin",
+					},
 				},
 			},
-		},
-		{
-			Query: "show create table a as of @Commit2;",
-			Expected: []sql.Row{
-				{"a", "CREATE TABLE `a` (\n" +
-					"  `pk` int NOT NULL,\n" +
-					"  `c1` int,\n" +
-					"  `c2` varchar(20),\n" +
-					"  PRIMARY KEY (`pk`)\n" +
-					") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin",
+			{
+				Query: "show create table a as of @Commit2;",
+				Expected: []sql.Row{
+					{"a", "CREATE TABLE `a` (\n" +
+						"  `pk` int NOT NULL,\n" +
+						"  `c1` int,\n" +
+						"  `c2` varchar(20),\n" +
+						"  PRIMARY KEY (`pk`)\n" +
+						") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin",
+					},
 				},
 			},
-		},
-		{
-			Query: "show create table a as of @Commit3;",
-			Expected: []sql.Row{
-				{"a", "CREATE TABLE `a` (\n" +
-					"  `pk` int NOT NULL,\n" +
-					"  `c2` varchar(20),\n" +
-					"  PRIMARY KEY (`pk`),\n" +
-					"  UNIQUE KEY `unique_c2` (`c2`)\n" +
-					") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin",
+			{
+				Query: "show create table a as of @Commit3;",
+				Expected: []sql.Row{
+					{"a", "CREATE TABLE `a` (\n" +
+						"  `pk` int NOT NULL,\n" +
+						"  `c2` varchar(20),\n" +
+						"  PRIMARY KEY (`pk`),\n" +
+						"  UNIQUE KEY `unique_c2` (`c2`)\n" +
+						") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin",
+					},
 				},
 			},
-		},
-		{
-			Query: "show create table a as of HEAD;",
-			Expected: []sql.Row{
-				{"a", "CREATE TABLE `a` (\n" +
-					"  `pk` int NOT NULL,\n" +
-					"  `c2` varchar(20),\n" +
-					"  PRIMARY KEY (`pk`),\n" +
-					"  UNIQUE KEY `unique_c2` (`c2`)\n" +
-					") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin",
+			{
+				Query: "show create table a as of HEAD;",
+				Expected: []sql.Row{
+					{"a", "CREATE TABLE `a` (\n" +
+						"  `pk` int NOT NULL,\n" +
+						"  `c2` varchar(20),\n" +
+						"  PRIMARY KEY (`pk`),\n" +
+						"  UNIQUE KEY `unique_c2` (`c2`)\n" +
+						") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin",
+					},
 				},
 			},
 		},
 	},
+	{
+		// "https://github.com/dolthub/dolt/issues/5478"
+		Name: "show table for default types with unique indexes",
```

Comment:
Keyless and unique keyless indexes have distinct implementations. We can ship even if those fail, but it would be useful to have skipped tests in that case.

Reply:
here are the function defaults tested in `TestColumnDefaults`
https://github.com/dolthub/go-mysql-server/blob/6ed5b50a67bec31be01a2263d43c1eff67b35321/enginetest/enginetests.go#L5954